invenio_oauthclient/contrib/orcid.py

```diff
@@ -144,47 +144,50 @@ def disconnect_handler(remote, *args, **kwargs):
     if orcid:
         oauth_unlink_external_id(dict(id=orcid, method='orcid'))
     if account:
-        account.delete()
+        with db.session.begin_nested():
+            account.delete()
 
-    return redirect(url_for('oauthclient_settings.index'))
+    return redirect(url_for('invenio_oauthclient_settings.index'))
 
 
-@session_manager
 def account_setup(remote, token, resp):
     """Perform additional setup after user have been logged in."""
     from invenio_oauthclient.utils import oauth_link_external_id
-    from invenio_ext.sqlalchemy import db
-
-    # Retrieve ORCID from response.
-    orcid = resp.get("orcid")
-
-    # Set ORCID in extra_data.
-    token.remote_account.extra_data = {"orcid": orcid}
-    user = token.remote_account.user
-
-    # Create user <-> external id link.
-    oauth_link_external_id(user, dict(id=orcid, method="orcid"))
-
-    # Fill user full name if not already set
-    if user and not any([user.given_names, user.family_name]):
-        # Query ORCID to get the real name
-        response = remote.get("{0}/orcid-bio".format(orcid),
-                              headers={'Accept': 'application/orcid+json'},
-                              content_type="application/json")
-
-        if response.status == 200:
-            try:
-                name = response.data["orcid-profile"]["orcid-bio"][
-                    "personal-details"]
-                user.given_names = name["given-names"]["value"]
-                user.family_name = name["family-name"]["value"]
-            except KeyError:
-                current_app.logger.exception(
-                    "Unexpected return format from ORCID: {0}".format(
-                        repr(response.data)))
-                return
-
-            db.session.add(user)
-
-            # Refresh user cache
-            current_user.reload()
+    from invenio_db import db
+
+    with db.session.begin_nested():
+        # Retrieve ORCID from response.
+        orcid = resp.get("orcid")
+
+        # Set ORCID in extra_data.
+        token.remote_account.extra_data = {"orcid": orcid}
+        user = token.remote_account.user
+
+        # Create user <-> external id link.
+        oauth_link_external_id(user, dict(id=orcid, method="orcid"))
+
+        # FIXME put these data in user profile!
```

Comment:
Can you please open a ticket for solving this issue so we don't forget. Right now the user is not created so I don't think ORCID login works.